apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -596,13 +585,14 @@ do_flush(
             %% must ensure the async worker is being monitored for
             %% such requests.
             IsUnrecoverableError = is_unrecoverable_error(Result),
+            WorkerPid = self(),
             case is_async_return(Result) of
                 true when IsUnrecoverableError ->
-                    ack_inflight(InflightTID, Ref, Id, Index);
+                    ack_inflight(InflightTID, Ref, Id, Index, WorkerPid);
                 true ->
                     ok;
                 false ->
-                    ack_inflight(InflightTID, Ref, Id, Index)
+                    ack_inflight(InflightTID, Ref, Id, Index, WorkerPid)
             end,
             {Data1, WorkerMRef} = ensure_async_worker_monitored(Data0, Result),
```

Comment:
Q: In `ensure_async_worker_monitored` we also have `WorkerPid`s, couldn't be that confusing?

Reply:
Good point, there are workers all around :sweat_smile: 
I think the differentiating point is that we have _buffer_ workers (the current module) and _async_ workers (from the connectors).  I'll add the `Async` modifier to the variable name to help with that differentiation.

Reply:
Fixed here: https://github.com/emqx/emqx/pull/10740